tests/descriptors/analysis_file/cf5d9300-3b81-52eb-a02f-25fb1364419e_2021-01-01T00:00:00.000000Z.json

```diff
@@ -0,0 +1,14 @@
+{
+    "content_type": "application/gzip; dcp-type=data",
+    "crc32c": "aff4c924",
+    "describedBy": "https://schema.staging.data.humancellatlas.org/system/2.0.0/file_descriptor",
+    "file_id": "cf5d9300-3b81-52eb-a02f-25fb1364419e",
+    "file_name": "AP1_file.h5ad",
+    "file_version": "2021-01-01T00:00:00.000000Z",
+    "s3_etag": "d04e9250c97e410e2749b93635595cc6",
+    "schema_type": "file_descriptor",
+    "schema_version": "2.0.0",
+    "sha1": "4c7e22be4a2bdc4571d1a836cb1a1757d40363f1",
+    "sha256": "8908aec2bf49fdc557a5841352c52c8af857707c6f9c760ff51fb6e0858a6577",
+    "size": 143
+}
```

Comment:
It would be nice to have a newline character at the end of the JSON so we don't always get this warning.

Reply:
And by warning I mean this one:

![image](https://user-images.githubusercontent.com/5143256/113643219-a8d71380-9636-11eb-88cc-4334979fe535.png)

Reply:
Thanks for the suggestion! We will modify the post-processor script to add the newline

Reply:
json.dumps doesn't add a trailing new line. We could add it, but just out of curiosity, is it a good practice to add a newline at the end of the file?

Reply:
read a bit on this https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file , https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline/729795 it's very unlikely that it will cause us problems but yeah i guess we can just add it

Reply:
@hannes-ucsc we have fixed the issue with the trailing newline, would you prefer for us to re-generate the data or would a third commit + squash work?

Reply:
If this were the only issue with the PR then a 3rd commit would be fine. But there was also

> The second commit adds a links.json but also renames the links.json from the first commit. Why does it do that? Shouldn't the second commit simply add a links.json and leave the other one as is? I worry that we have a non-determinism in the UUID generation for the links.json files.

which looks like @aaclan-ebi already addressed but we should verify her fix works. Ideally, you would reset this branch to master and then run the tooling twice as (I assume) you did the first time around. With the fix for the links ID non-determinism we should not observe the second commit to touch the links.json added by the first commit, correct?

Reply:
Yes, you are correct!